docs/odh/authorino-setup-manual.md

```diff
@@ -0,0 +1,155 @@
+   the following resource:
+    ```yaml
+    apiVersion: maistra.io/v1
+    kind: ServiceMeshMember
+    metadata:
+      name: default
+      namespace: opendatahub-auth-provider
+    spec:
+      controlPlaneRef:
+        namespace: istio-system
+        name: data-science-smcp
+    ```
+1. Create the following `Authorino` resource:
+    ```yaml
+    apiVersion: operator.authorino.kuadrant.io/v1beta1
+    kind: Authorino
+    metadata:
+      name: authorino
+      namespace: opendatahub-auth-provider
+    spec:
+      authConfigLabelSelectors: security.opendatahub.io/authorization-group=default
+      clusterWide: true
+      listener:
+        tls:
+          enabled: false
+      oidcServer:
+        tls:
+          enabled: false
+    ```
+1. Once Authorino is running, patch the Authorino deployment to inject the Istio
+sidecar and make it part of the Service Mesh:
+    * `oc patch deployment authorino -n opendatahub-auth-provider -p '{"spec": {"template":{"metadata":{"labels":{"sidecar.istio.io/inject":"true"}}}} }'`
+
+## Prepare the Service Mesh to work with Authorino
+
+Once the Authorino instance is configured, the `ServiceMeshControlPlane` resource
+needs to be modified to configure Authorino as an authorization provider of the Mesh.
+
+Assuming you have your `ServiceMeshControlPlane` resource in the `istio-system` namespace,
+you named it `data-science-smcp`, and you followed the previous section without
+renaming any resource, the following is a patch that you can apply with
+`oc patch smcp --type merge -n istio-system`:
+
+```yaml
+spec:
+  techPreview:
+    meshConfig:
+      extensionProviders:
+      - name: opendatahub-auth-provider
+        envoyExtAuthzGrpc:
+          service: authorino-authorino-authorization.opendatahub-auth-provider.svc.cluster.local
+          port: 50051
+```
+
+> [!IMPORTANT]
+> You should apply this patch only if you don't have other extension providers. If you do,
+> you should manually edit your `ServiceMeshControlPlane` resource and add the
+> needed configuration.
+
+## Configure KServe authorization
+
+In ODH KServe, authorization is configured using a global Istio
+`AuthorizationPolicy` targeting the predictor pods of InferenceServices. Also,
+given the several hops of a request, an `EnvoyFilter` is used to reset the HTTP
+_Host_ header to the original one of the inference request.
+
+The following YAML are the resources that you need to apply to the namespace
+where Service Mesh is installed, assuming you have followed the previous
+instructions without renaming any resource. If you created your `ServiceMeshControlPlane`
+resource in the `istio-system` namespace, you can create the resources using
+an `oc apply -n istio-system` command.
+
+```yaml
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+   name: kserve-predictor
+spec:
+   action: CUSTOM
+   provider:
+      name: opendatahub-auth-provider
```

Comment:
Where can I find this envoyfilter ?

Reply:
It is not an envoyfilter, it is a configuration in the ServiceMeshControlPlane. See: https://github.com/opendatahub-io/kserve/pull/253/files#diff-649512b8d7a6b58f996c1a23ee4e257cf224998c7ef287cc3b722b8f8c07ddcbR71